frontends/web/src/routes/account/summary/chart.tsx

```diff
@@ -28,141 +29,195 @@ import { DefaultCurrencyRotator } from '../../../components/rates/rates';
+
+  const [source, setSource] = useState<'daily' | 'hourly'>('daily');
+  const [difference, setDifference] = useState<number>();
+  const [diffSince, setDiffSince] = useState<string>();
+  const [tooltipData, setTooltipData] = useState<{
+    toolTipVisible: boolean;
+    toolTipValue?: string;
+    toolTipTop: number;
+    toolTipLeft: number;
+    toolTipTime: number;
+  }>({
     toolTipVisible: false,
-    toolTipValue: undefined,
     toolTipTop: 0,
     toolTipLeft: 0,
     toolTipTime: 0,
-    isMobile: false,
-  };
+  });
+
+  const prevChartDataDaily = usePrevious(data.chartDataDaily);
+  const prevChartDataHourly = usePrevious(data.chartDataHourly);
+  const prevChartFiat = usePrevious(data.chartFiat);
+  const prevHideAmounts = usePrevious(hideAmounts);
 
-  public componentDidMount() {
-    this.createChart();
+  if (source === 'daily' && prevChartDataDaily?.length !== data.chartDataDaily.length) {
+    lineSeries.current?.setData(data.chartDataDaily);
+    chart.current?.timeScale().fitContent();
+    setFormattedData(data.chartDataDaily);
   }
 
-  public componentWillUnmount() {
-    window.removeEventListener('resize', this.onResize);
-    if (this.chart) {
-      this.chart.timeScale().unsubscribeVisibleLogicalRangeChange(this.calculateChange);
-      this.chart.unsubscribeCrosshairMove(this.handleCrosshair);
-    }
+  if (source === 'hourly' && prevChartDataHourly?.length !== data.chartDataHourly.length) {
+    lineSeries.current?.setData(data.chartDataHourly);
+    chart.current?.timeScale().fitContent();
+    setFormattedData(data.chartDataHourly);
   }
 
-  public componentDidUpdate(prev: Props) {
-    const { chartDataDaily, chartDataHourly } = this.props.data;
-    if (!this.chart) {
-      this.createChart();
+  if (prevChartFiat !== data.chartFiat) {
+    reinitializeChart();
+  }
+
+  if (prevHideAmounts !== hideAmounts) {
+    chart.current?.applyOptions({
+      leftPriceScale: {
+        visible: hideAmounts ? false : !isMobile,
+      }
+    });
+  }
+
+  useEffect(() => {
+    initChart();
+    return () => {
+      window.removeEventListener('resize', onResize);
+      if (chart.current) {
+        chart.current.timeScale().unsubscribeVisibleLogicalRangeChange(calculateChange);
+        chart.current.unsubscribeCrosshairMove(handleCrosshair);
+      }
+    };
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (!chart.current) {
+      initChart();
     }
-    if (
-      (this.lineSeries && prev.data.chartDataDaily && prev.data.chartDataHourly && chartDataDaily && chartDataHourly)
-        && (
-          prev.data.chartDataDaily.length !== chartDataDaily.length
-          || prev.data.chartDataHourly.length !== chartDataHourly.length
-        )
-    ) {
-      const data = this.state.source === 'hourly' ? chartDataHourly : chartDataDaily;
-      this.lineSeries.setData(data);
-      this.chart?.timeScale().fitContent();
-      this.setFormattedData(data);
+  });
```

Comment:
I believe this effect is not needed, as the previous effect already and unconditionally calls initChart().

Reply:
yes, I also thought that, but I thought I should put it in there since there was [this](https://github.com/BitBoxSwiss/bitbox-wallet-app/pull/2728/files#diff-c761d5505dba4edd4496dfe892f927231782977bae81a155d645a6a04dd7ab59L109).

So yes logically it does not need to be there but if we want to strictly say we only want to convert the component it should be there? But I'll remove it and test if everything still works.